perun-core/src/test/java/cz/metacentrum/perun/core/entry/ConsentsManagerEntryIntegrationTest.java

```diff
@@ -620,6 +639,282 @@ public void evaluateConsentsWithInsufficientUnsignedConsent() throws Exception {
 		assertThat(userConsents.get(0).getStatus()).isEqualTo(ConsentStatus.UNSIGNED);
 	}
 
+	@Test
+	public void evaluateConsentsForConsentHubCreatesConsents() throws Exception {
+		System.out.println(CLASS_NAME + "evaluateConsentsForConsentHubCreatesConsents");
+
+		User user1 = setUpUser("Harry", "Doe");
+		User user2 = setUpUser("James", "Doe");
+		Member member1 = perun.getMembersManager().createMember(sess, vo, user1);
+		Member member2 = perun.getMembersManager().createMember(sess, vo, user2);
+		perun.getGroupsManagerBl().addMember(sess, group, member1);
+		perun.getGroupsManagerBl().addMember(sess, group, member2);
+		perun.getGroupsManager().setMemberGroupStatus(sess, member1, group, MemberGroupStatus.EXPIRED);
+		perun.getGroupsManager().setMemberGroupStatus(sess, member2, group, MemberGroupStatus.VALID);
+
+		// validate both members in Vo, otherwise they will be skipped
+		perun.getMembersManagerBl().validateMember(sess, member1);
+		perun.getMembersManagerBl().validateMember(sess, member2);
+
+		System.out.println("members:\n" + perun.getGroupsManagerBl().getGroupMembers(sess, group));
+
+
+
+		ConsentHub consentHub = consentsManagerEntry.getConsentHubByFacility(sess, facility.getId());
+
+		boolean originalForce = BeansUtils.getCoreConfig().getForceConsents();
+		try {
+			BeansUtils.getCoreConfig().setForceConsents(true);
+			perun.getConsentsManagerBl().evaluateConsents(sess, consentHub);
+		} finally {
+			BeansUtils.getCoreConfig().setForceConsents(originalForce);
+		}
+
+		List<Consent> consents = consentsManagerEntry.getConsentsForUserAndConsentHub(sess, user2.getId(), consentHub.getId());
+		List<Consent> shouldBeEmptyConsents = consentsManagerEntry.getConsentsForUserAndConsentHub(sess, user1.getId(), consentHub.getId());
+
+		Consent expectedConsent = new Consent(consents.get(0).getId(), user2.getId(), consentHub, List.of(attrDef));
+		assertThat(consents).containsExactly(expectedConsent);
+		assertEquals(0, shouldBeEmptyConsents.size());
+
+		// evaluate again, still should have only the one UNSIGNED consent
+		try {
+			BeansUtils.getCoreConfig().setForceConsents(true);
+			perun.getConsentsManagerBl().evaluateConsents(sess, consentHub);
+		} finally {
+			BeansUtils.getCoreConfig().setForceConsents(originalForce);
+		}
+		assertThat(consents).containsExactly(expectedConsent);
```

Comment:
We should probably retrieve the consents list again after the second evaluation.